xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.h

```diff
@@ -104,6 +105,7 @@ class CProcessorHD : public ID3DResource
   D3D11_VIDEO_PROCESSOR_RATE_CONVERSION_CAPS m_rateCaps;
   D3D11_TEXTURE2D_DESC m_texDesc;
   PROCESSOR_VIEW_TYPE m_eViewType;
+  std::vector<AVPixelFormat> m_formats;
```

Comment:
>  if I don't know will it be used or not

this problem exists in current master.

in this branch a decoder delivers a videobuffer and renderer decides if it can handle it. capabilites don't change at runtime, means this is very static behaviour.  in case it depends on runtime environemnt, you can ask windowing in CProcessInfo::GetRenderFormats